apps/03TicTacToe.js

```diff
@@ -25,22 +26,60 @@ function printBoard() {
 
 function horizontalWin() {
   // Your code here
+  return  (board[0][0] === playerTurn && board[0][1] === playerTurn && board[0][2] === playerTurn) ||
+          (board[1][0] === playerTurn && board[1][1] === playerTurn && board[1][2] === playerTurn) ||
+          (board[2][0] === playerTurn && board[2][1] === playerTurn && board[2][2] === playerTurn);
 }
 
 function verticalWin() {
   // Your code here
+  return  (board[0][0] === playerTurn && board[1][0] === playerTurn && board[2][0] === playerTurn) ||
+          (board[0][1] === playerTurn && board[1][1] === playerTurn && board[2][1] === playerTurn) ||
+          (board[0][2] === playerTurn && board[1][2] === playerTurn && board[2][2] === playerTurn);
 }
 
 function diagonalWin() {
   // Your code here
+  return  (board[0][0] === playerTurn && board[1][1] === playerTurn && board[2][2] === playerTurn) ||
+          (board[0][2] === playerTurn && board[1][1] === playerTurn && board[2][0] === playerTurn);
 }
 
 function checkForWin() {
   // Your code here
+  if (horizontalWin() || verticalWin() || diagonalWin()) {
+    console.log('Player ' + playerTurn + ' Won!');
+    stopGame = true;
+    return true;
+  }
+}
+
+function checkForTie() {
+  var isEmpty = false;
+  for(var i = 0; i < board.length; i++) {
+    for(var j = 0; j < board[i].length; j++) {
+      isEmpty = (board[i][j] === ' ') ? true : false;
+      if (isEmpty) {
+        return;
```

Comment:
I didn't want to return anything. If it found an empty square I wanted it to stop looking and end the function. I was overthinking it.